### PR TITLE
fix some crashes under Catalina

### PR DESCRIPTION
**Pull request by skurfer**
This should force some lingering UI things onto the main thread. These have been warnings for a while, but under Catalina, they will crash QS.

Progress toward #2497

### CODE REVIEW

**Comment by skurfer** [comment]
Any objections to how this was handled @tiennou @pjrobertson ?

I had originally changed the code that posts the notification that triggers `catalogIndexingFinished:`, but in the end, I decided that code should be on the main thread no matter how it’s called.

File: Quicksilver/Code-QuickStepInterface/QSObjectView.m [hunk 1]
```diff
@@ -182,14 +182,21 @@ - (void)setPreviousObjectValue:(QSObject *)aValue
   previousObjectValue = aValue;
 }
 
-- (id)objectValue { return [[self cell] representedObject];  }
+- (id)objectValue {
+	__block id objVal;
+	QSGCDMainSync(^{
```
Comment (tiennou) [inline Quicksilver/Code-QuickStepInterface/QSObjectView.m]: That's hacky as possible, but… yeah. The argument I'm going through when trying to fix those kind of issues is mostly this :

> does the "function" performed by the code touches AppKit objects, (ie views) ? does it benefit from being able to be multi-threaded ?

In that case, this is a view, so there should be no need to dispatch anything to the main thread — you *are* in the view's code, so it's whatever controller-level background-receiving notification-handling code that caused that call to happen that needs some proxying. OTOH, those thread-safety issues are *everywhere*, so if that fixes most of the crashes, I'm fine with 🚢.

Comment (tiennou) [reply Quicksilver/Code-QuickStepInterface/QSObjectView.m]: Thanks for doing that @skurfer. I don't intend to sound harsh in any way shape or form, but…

> whatever controller-level background-receiving notification-handling code

could call any other AppKit object, and you'd still get a crash.

I do have a non-dying MacBook Pro again, so hopefully I'll be able to actually *open* the QS project, and help with some of the issues/PRs/cleanups.

Comment (skurfer) [reply Quicksilver/Code-QuickStepInterface/QSObjectView.m]: I did put a break point in there to see where `objectValue` was being called from so I could just make sure all of that stuff was on the main thread instead, but like you said, it’s _everywhere_.

**Pull request merged**